Sources/joyfill-components-swift/Components/DateTime/RPicker.swift

```diff
@@ -0,0 +1,461 @@
+                    }
+                }
+            } else {
+                //restrict to use wheel mode
+                datePickerStyle = .Wheel
+                height = height + pickerHeight
+            }
+        } else {
+            stackView.addArrangedSubview(optionPicker)
+            height = height + pickerHeight
+        }
+        
+        self.view.addSubview(stackView)
+        stackView.pinConstraints(view, left: 0, right: 0, bottom: 0, height: height)
+        
+        if #available(iOS 12.0, *) {
+            if traitCollection.userInterfaceStyle == .dark {
+                setUpThemeMode(isDark: true)
+            } else {
+                setUpThemeMode(isDark: false)
+            }
+        } else {
+            // Fallback on earlier versions
+            setUpThemeMode(isDark: false)
+        }
+    }
+    
+    private func setUpThemeMode(isDark: Bool) {
+        if isDark {
+            titleLabel.textColor = UIColor.white
+            stackView.backgroundColor = UIColor.black
+            transView.backgroundColor = UIColor(white: 1, alpha: 0.3)
+            if pickerType == .date {
+                datePicker.backgroundColor = UIColor.black
+            } else {
+                optionPicker.backgroundColor = UIColor.black
+            }
+            toolBarView.backgroundColor = UIColor.black
+            
+        } else {
+            titleLabel.textColor = UIColor.darkGray
+            stackView.backgroundColor = UIColor.white
+            transView.backgroundColor = UIColor(white: 0.1, alpha: 0.3)
+            if pickerType == .date {
+                datePicker.backgroundColor = UIColor.white
+            } else {
+                optionPicker.backgroundColor = UIColor.white
+            }
+            toolBarView.backgroundColor = UIColor.white
+        }
+    }
+    
+    private func dismissVC() {
+        onWillDismiss?()
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func handleTap() { dismissVC() }
+    
+    //MARK:- Private properties
+    private lazy var transView: UIView = {
+        let vw = UIView()
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(self.handleTap))
+        vw.addGestureRecognizer(tapGesture)
+        vw.isUserInteractionEnabled = true
+        return vw
+    }()
+    
+    private lazy var stackView: UIStackView = {
+        let sv = UIStackView()
+        sv.axis = NSLayoutConstraint.Axis.vertical
+        sv.distribution = UIStackView.Distribution.fill
+        sv.alignment = UIStackView.Alignment.center
+        sv.spacing = 0.0
+        return sv
+    }()
+    
+    private lazy var datePicker: UIDatePicker = {
+        let picker = UIDatePicker()
+        picker.pinConstraints(view, width: view.frame.width)
+        picker.minimumDate = minDate
+        picker.maximumDate = maxDate
+        picker.date = selectedDate
+        picker.datePickerMode = datePickerMode
+        
+        if #available(iOS 14, *) {
```

Comment:
Are we testing this on simulators that aren't iOS 14? Usually if there is any version specific code you'd want to test that version but devices that aren't that version

Reply:
It's a iOS version check as if a device with lower ios version that 14 will run with default preferredDatePickerStyle (typically wheels).